ogc/src/main/scala/geotrellis/server/ogc/OgcTime.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2020 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.server.ogc
+
+import jp.ne.opt.chronoscala.Imports._
+import cats.data.NonEmptyList
+import cats.{Order, Semigroup}
+import cats.implicits._
+
+import java.time.ZonedDateTime
+
+sealed trait OgcTime {
+  def isEmpty: Boolean = false
+  def nonEmpty: Boolean = !isEmpty
+}
+
+case object OgcTimeEmpty extends OgcTime {
+  override val isEmpty: Boolean = true
+  override val nonEmpty: Boolean = !isEmpty
+
+  implicit val ogcTimeEmptySemigroup: Semigroup[OgcTimeEmpty.type] = Semigroup.instance { (l, _) => l }
+}
+
+object OgcTime {
+  implicit val ogcTimeSemigroup: Semigroup[OgcTime] = Semigroup.instance {
+    case (l: OgcTimePositions, r: OgcTimePositions) => l |+| r
+    case (l: OgcTimeInterval, r: OgcTimeInterval) => l |+| r
+    case (l: OgcTimePositions, _: OgcTimeEmpty.type) => l
+    case (l: OgcTimeInterval, _: OgcTimeEmpty.type) => l
+    case (_: OgcTimeEmpty.type, r: OgcTimePositions) => r
+    case (_: OgcTimeEmpty.type, r: OgcTimeInterval) => r
+    case (l, _) => l
+  }
+}
+
+final case class OgcTimePositions(list: NonEmptyList[ZonedDateTime]) extends OgcTime {
+  override def toString: String = list.toList.map(_.toInstant.toString).mkString(", ")
+}
+
+object OgcTimePositions {
+  implicit val timeOrder: Order[ZonedDateTime] = { (l, r) => implicitly[Ordering[ZonedDateTime]].compare(l, r) }
+
+  implicit val ogcTimePositionsSemigroup: Semigroup[OgcTimePositions] = Semigroup.instance { (l, r) =>
+    OgcTimePositions((l.list ::: r.list).distinct)
+  }
+
+  def apply(timePeriod: ZonedDateTime): OgcTimePositions = OgcTimePositions(NonEmptyList(timePeriod, Nil))
+  def apply(timeString: String): OgcTimePositions = apply(ZonedDateTime.parse(timeString))
+  def apply(times: List[ZonedDateTime]): OgcTime =
+    times match {
+      case head :: tail => OgcTimePositions(NonEmptyList(head, tail))
+      case _ => OgcTimeEmpty
+    }
+  def apply(times: List[String])(implicit d: DummyImplicit): OgcTime = apply(times.map(ZonedDateTime.parse))
+}
+
+/**
+  * Represents the TimeInterval and TimePosition types used in TimeSequence requests
```

Comment:
Docstring should be updated to remove refs to `TimePosition` now that this explicitly represents an interval.